File: src/main/java/toast/pine/Entities/MonsterListener.java
package toast.pine.Entities;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import toast.pine.Events.MonsterDeathEvent;
import toast.pine.Events.MonsterSpawnEvent;
import toast.pine.Events.MonsterTargetPlayerEvent;

public class MonsterListener implements Listener {

    private final EntityManager manager;

    public MonsterListener(EntityManager manager) {
        this.manager = manager;
    }


    @EventHandler
    public void onMonsterSpawn(MonsterSpawnEvent event) {
        this.manager.handleMonsterSpawn(event);
    }

    @EventHandler
    public void onMonsterDeath(MonsterDeathEvent event){
        this.manager.handleMonsterDeath(event);
    }

    @EventHandler
    public void onMonsterTarget(MonsterTargetPlayerEvent event){
        this.manager.handlerMonsterTarget(event);
    }

}

File: src/main/java/pine/toast/toastrpg/monsters/MonsterListener.kt
package pine.toast.toastrpg.monsters

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import pine.toast.toastrpg.ToastRPG
import pine.toast.toastrpg.entities.EntityManager
import pine.toast.toastrpg.events.MonsterDeathEvent
import pine.toast.toastrpg.events.MonsterSpawnEvent
import pine.toast.toastrpg.events.MonsterTargetPlayerEvent

class MonsterListener(manager: EntityManager) : Listener {
    private val manager: EntityManager

    init {
        this.manager = manager
    }

    @EventHandler
    fun onMonsterSpawn(event: MonsterSpawnEvent) {
        ToastRPG.getMonsterFactory()!!.handleMonsterSpawn(event)
    }

    @EventHandler
    fun onMonsterDeath(event: MonsterDeathEvent) {
        ToastRPG.getMonsterFactory()!!.handleMonsterDeath(event)
    }

    @EventHandler
    fun onMonsterTarget(event: MonsterTargetPlayerEvent) {
        ToastRPG.getMonsterFactory()!!.handlerMonsterTarget(event)
    }
}
